cpmpy/tools/explain/mus.py

```diff
@@ -184,3 +236,85 @@ def do_recursion(soft, hard, delta):
 
     core = do_recursion(soft, hard, [])
     return core
+
+def optimal_mus_naive(soft, hard=[], weights=1, solver="ortools", hs_solver="ortools", grow=mss_grow_naive):
+    """
+        Naive implementation of `optimal_mus` without assumption variables and incremental solving
+    """
+
+    soft = toplevel_list(soft, merge_and=False)
+    bvs = cp.boolvar(shape=len(soft))
+
+    if is_num(weights):
+        weights = np.ones(len(bvs), dtype=int)
+    hs_solver = cp.SolverLookup.get(hs_solver)
+    hs_solver.minimize(cp.sum(bvs * np.array(weights)))
+
+    while hs_solver.solve() is True:
+
+        hitting_set = [c for bv, c in zip(bvs, soft) if bv.value()]
+        if cp.Model(hard + hitting_set).solve(solver=solver) is False:
+            break
+        # SAT, how to do grow without assumptions?
```

Comment:
I'm confused... we don't grow in the above version either? though we could...
I think to mimic the above you just have to copy the disjoint part?

Reply:
We do implicitely grow in the assumption version, by looking at the assignment found  by the solver.
I.e., there may be more assumption set to True, besides `hitting_set`. Actually, that's the premise of why the correction-subset enumeration works in that algo.
But here, we don't get any "free" extra constraints in the sat_subset